pyspike/PieceWiseConstFunc.py

```diff
@@ -129,19 +129,22 @@ def integral(self, interval=None):
             # no interval given, integrate over the whole spike train
             a = np.sum((self.x[1:]-self.x[:-1]) * self.y)
         else:
-            # find the indices corresponding to the interval
-            start_ind = np.searchsorted(self.x, interval[0], side='right')
-            end_ind = np.searchsorted(self.x, interval[1], side='left')-1
-            assert start_ind > 0 and end_ind < len(self.x), \
-                "Invalid averaging interval"
-            # first the contribution from between the indices
-            a = np.sum((self.x[start_ind+1:end_ind+1] -
-                        self.x[start_ind:end_ind]) *
-                       self.y[start_ind:end_ind])
-            # correction from start to first index
-            a += (self.x[start_ind]-interval[0]) * self.y[start_ind-1]
-            # correction from last index to end
-            a += (interval[1]-self.x[end_ind]) * self.y[end_ind]
```

Comment:
BTW I'm sure I could have found a way to fix this implementation, but I couldn't get my head around it, so re-wrote it using a slightly different approach.

Reply:
I think all that is needed is to handle the special case where start_index > end_index, which happens when the given interval is between values.